features/earn/aave/open/state/openAaveStateMachine.ts

```diff
@@ -0,0 +1,149 @@
+import { Observable } from 'rxjs'
+import { assign, createMachine, spawn } from 'xstate'
+
+import { TokenBalances } from '../../../../../blockchain/tokens'
+import { zero } from '../../../../../helpers/zero'
+import { ProxyStateMachine } from '../../../../proxyNew/state/proxyStateMachine.types'
+import { getNameOfService, services } from './openAaveStateMachine.services'
+import { OpenAaveContext, OpenAaveEvent, OpenAaveState } from './openAaveStateMachine.types'
+
+export function openAaveStateMachine(
+  tokenBalances$: Observable<TokenBalances>,
+  proxyAddress$: Observable<string | undefined>,
+  getProxyStateMachine: () => ProxyStateMachine,
+) {
+  return createMachine<OpenAaveContext, OpenAaveEvent, OpenAaveState>(
+    {
+      key: 'aaveOpen',
+      initial: 'editing',
+      context: {
+        dependencies: {
+          tokenBalances$,
+          getProxyStateMachine,
+          proxyAddress$,
+        },
+      },
+      states: {
+        editing: {
+          entry: [
+            assign((context) => ({
+              currentStep: 1,
+              token: 'ETH',
+              totalSteps: context.proxyAddress ? 2 : 3,
+              canGoToNext: context.proxyAddress === undefined || context.amount?.gt(zero),
+            })),
+          ],
+          invoke: {
+            src: getNameOfService('initMachine'),
+          },
+          on: {
+            SET_BALANCE: {
+              actions: [
+                assign((_, event) => ({
+                  tokenBalance: event.balance,
+                  tokenPrice: event.tokenPrice,
+                })),
+              ],
+            },
+            PROXY_ADDRESS_RECEIVED: {
+              actions: [
+                assign((_, event) => ({
+                  proxyAddress: event.proxyAddress,
+                  totalSteps: event.proxyAddress ? 2 : 3,
+                })),
+              ],
+            },
+            SET_AMOUNT: {
+              actions: [
+                assign((context, event) => ({
+                  amount: event.amount,
+                  auxiliaryAmount: event.amount.times(context.tokenPrice || zero),
+                  canGoToNext: context.proxyAddress === undefined || event.amount.gt(0),
+                })),
+              ],
+            },
+            CREATE_PROXY: {
+              target: 'proxyCreating',
+              cond: (context, _) => context.proxyAddress === undefined,
+            },
+            CONFIRM_DEPOSIT: {
+              target: 'reviewing',
+              cond: (context, _) => context.proxyAddress !== undefined,
```

Comment:
Could we use multiple guards to this event so

https://xstate.js.org/docs/guides/guards.html#multiple-guards

```suggestion
              cond: (context, _) => context.proxyAddress !== undefined && context.amount.gt(0),
```